propagation/http_trace_context_propagator_benchmark_test.go

```diff
@@ -0,0 +1,89 @@
+package propagation
+
+import (
+	"context"
+	"net/http"
+	"testing"
+
+	"go.opentelemetry.io/api/core"
+	"go.opentelemetry.io/api/trace"
+	mocktrace "go.opentelemetry.io/internal/trace"
+)
+
+func BenchmarkStartEndSpan(b *testing.B) {
+	t := httpTraceContextPropagator{}
+
+	injectSubBenchmark(b, func(ctx context.Context, b *testing.B) {
+		req, _ := http.NewRequest("GET", "http://example.com", nil)
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			t.Inject(ctx, req.Header)
+		}
+	})
+}
+
+func injectSubBenchmark(b *testing.B, fn func(context.Context, *testing.B)) {
+	b.Run("SampledSpanContext", func(b *testing.B) {
+		var (
+			id      uint64
+			spanID  = uint64(0x00f067aa0ba902b7)
+			traceID = core.TraceID{High: 0x4bf92f3577b34da6, Low: 0xa3ce929d0e0e4736}
+		)
+		mockTracer := &mocktrace.MockTracer{
+			Sampled:     false,
+			StartSpanId: &id,
+		}
+		b.ReportAllocs()
+		sc := core.SpanContext{
+			TraceID:    traceID,
+			SpanID:     spanID,
+			TraceFlags: core.TraceFlagsSampled,
+		}
+		ctx := context.Background()
+		ctx, _ = mockTracer.Start(ctx, "inject", trace.ChildOf(sc))
+		fn(ctx, b)
+	})
+
+	b.Run("WithoutSpanContext", func(b *testing.B) {
+		b.ReportAllocs()
+		ctx := context.Background()
+		fn(ctx, b)
+	})
+}
+
+func BenchmarkExtract(b *testing.B) {
+	extractSubBenchmark(b, func(b *testing.B, req *http.Request) {
+		propagator := HttpTraceContextPropagator()
+		ctx := context.Background()
+		b.ResetTimer()
+		for i := 0; i < b.N; i++ {
+			propagator.Extract(ctx, req.Header)
+		}
+	})
+}
+
+func extractSubBenchmark(b *testing.B, fn func(*testing.B, *http.Request)) {
+	trace.SetGlobalTracer(&mocktrace.MockTracer{})
```

Comment:
Please try not to use the global tracer in testing. There will be a future change that only allows the global tracer to be set once per process lifetime.

Reply:
I fixed the global tracer and added the results of this benchmark.